source/common/buffer/buffer_impl.cc

```diff
@@ -555,6 +555,26 @@ OwnedImpl::OwnedImpl(const Instance& data) : OwnedImpl() { add(data); }
 
 OwnedImpl::OwnedImpl(const void* data, uint64_t size) : OwnedImpl() { add(data, size); }
 
+OwnedImpl::OwnedImpl(const OwnedImpl& other) : OwnedImpl() { add(other); }
+
+OwnedImpl::OwnedImpl(OwnedImpl&& other) : OwnedImpl() { move(other); }
+
+OwnedImpl& OwnedImpl::operator=(const OwnedImpl& other) {
```

Comment:
Having the copy constructor and copy assignment operators can potentially lead to unintentional copies through C++ magic, so I would prefer to not have them. Is it possible to hide these behind explicit methods so they are very intentional?

Reply:
I brought up the previous discussion about why we added copy contractor. Please refer to that thread.

Reply:
I think the previous state was a strange explicit upcast which had more or less then same effect.

How about an explicit copyFrom() method instead, which you could call from the constructor body. Would that work?

Reply:
Yes, +1 to an explicit method. I don't think we need the constructor/assignment overrides but could have explicit methods (possible static) which do what we need. you could also make these operators private, and then access them via public static factory methods.

/wait

Reply:
Buffer::Instance already has similar method: add(). I switch to that one.